jni/src/test/java/com/horizen/schnorrnative/SchnorrKeyPairTest.java

```diff
@@ -17,6 +17,36 @@ public void testGenerate() throws Exception {
         }
     }
 
+    @Test
+    public void testDeriveFromSeed() throws Exception {
+        byte[] seed = { 1, 2, 3, 4, 5, 6, 7, 8 };
+        byte[] expectedPubKeyBytes = {83, -1, 54, 52, 44, -74, -36, 113, 87, -126, 80, -84, -45, -116, 88, -69, 73, 118, -54, 100, 112, 80, 22, 64, 87, -93, 79, 2, -86, -48, 107, 31, 0};
+        byte[] expectedSecretKeyBytes = {41, 70, -14, -11, 47, 124, 108, -114, -83, -97, 44, -44, 54, 63, -98, 6, 45, 45, 40, 5, 85, -37, 98, 59, -100, 77, -90, -61, -123, -50, -16, 13};
+        try(SchnorrKeyPair keyPair = SchnorrKeyPair.generate(seed))
+        {
+            assertNotNull("Key pair derive from seed was unsuccessful", keyPair);
+            assertTrue("Public key verification failed.", keyPair.getPublicKey().verifyKey());
+            assertArrayEquals("Derive from seed didn't produce the expected public key", expectedPubKeyBytes, keyPair.getPublicKey().serializePublicKey());
+            assertArrayEquals("Derive from seed didn't produce the expected secret key", expectedSecretKeyBytes, keyPair.getSecretKey().serializeSecretKey());
+        }
+        byte[] emptySeed = {};
+        try(SchnorrKeyPair keyPair = SchnorrKeyPair.generate(seed))
+        {
+            assertNotNull("Key pair derive from empty seed was unsuccessful", keyPair);
+            assertTrue("Public key verification failed.", keyPair.getPublicKey().verifyKey());
+        }
+
+    }
+
+    private void assertArrayEquals(String message, byte[] expected, byte[] actual) {
```

Comment:
`assertArrayEquals` is a part of JUnit suite. 
Can be reused.
same in other test

Reply:
Except that it isn't. I thought so as well but then travis-CI couldn't compile it so I wrote it myself. See [Job 581609520](https://app.travis-ci.com/github/HorizenOfficial/zendoo-sc-cryptolib/jobs/581609520)

Reply:
Perhaps we are using an old version of JUnit ?
If not let's keep it, but can you move it please to the TestUtils class ?